src/query/api/v1/handler/prometheus/native/read.go

```diff
@@ -100,9 +127,13 @@ func (h *PromReadHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	w.Header().Set("Content-Type", "application/json")
 	if params.FormatType == models.FormatM3QL {
 		renderM3QLResultsJSON(w, result, params)
+		h.promReadMetrics.fetchSuccess.Inc(1)
+		timer.Stop()
```

Comment:
Nit: should probably just defer `timer.Stop()`

Reply:
so I don't want it to record unless it's a success.